main.go

```diff
@@ -440,7 +440,7 @@ func main() {
 
 		bv := func(collector otelv1beta1.OpenTelemetryCollector) admission.Warnings {
 			var warnings admission.Warnings
-			params, newErr := collectorReconciler.GetParams(collector)
+			params, newErr := collectorReconciler.GetParams(context.Background(), collector)
```

Comment:
Does the underlying implementation handle timeouts and all that stuff?

Reply:
It does, but now that you mention it, this should really be the context in which the caller runs. I'll modify the build validation function itself to take a context argument instead.

Reply:
Fixed.